mlrun/model_monitoring/application.py

```diff
@@ -82,43 +83,42 @@ def run_application(
                 latest_request: pd.Timestamp,
                 endpoint_id: str,
                 output_stream_uri: str,
-            ) -> Union[ModelMonitoringApplicationResult, list[ModelMonitoringApplicationResult]
-            ]:
+            ) -> ApplicationResult:
                 self.context.log_artifact(TableArtifact("sample_df_stats", df=sample_df_stats))
-                return ModelMonitoringApplicationResult(
+                return ApplicationResult(
                     name="data_drift_test",
                     value=0.5,
                     kind=mm_constant.ResultKindApp.data_drift,
-                    status = mm_constant.ResultStatusApp.detected,
-                    extra_data={})
+                    status=mm_constant.ResultStatusApp.detected,
+                )
 
         # mlrun: end-code
     """
 
     kind = "monitoring_application"
 
-    def do(
-        self, event: dict[str, Any]
-    ) -> Tuple[list[ModelMonitoringApplicationResult], dict]:
+    def do(self, event: dict[str, Any]) -> Tuple[list[ApplicationResult], dict]:
         """
         Process the monitoring event and return application results.
 
         :param event:   (dict) The monitoring event to process.
-        :returns:       (list[ModelMonitoringApplicationResult]) The application results.
+        :returns:       (list[ApplicationResult], dict) The application results
+                        and the original event for the application.
         """
         resolved_event = self._resolve_event(event)
         if not (
             hasattr(self, "context") and isinstance(self.context, mlrun.MLClientCtx)
         ):
             self._lazy_init(app_name=resolved_event[0])
-        results = self.run_application(*resolved_event)
+        results = self.run(*resolved_event)
         results = results if isinstance(results, list) else [results]
         return results, event
 
     def _lazy_init(self, app_name: str):
         self.context = self._create_context_for_logging(app_name=app_name)
 
-    def run_application(
+    @abstractmethod
+    def run(
```

Comment:
I don't see any benefits to this change only can make a confusion to the `run()` of the nuclio function or the `run()` method of the project class.

Also you can see [here](https://jira.iguazeng.com/browse/ML-5295?focusedCommentId=121022&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-121022) amit's comment, and according to it maybe the right name can be 'analyze' or something like that.

Reply:
a34682e